### PR TITLE
Add Power8 to list of CPUs that support unaligned memory access

### PR DESCRIPTION
**Pull request by chrisvest**
Also add small cleanups for CommonAbstractStore, and a fix for a flaky test.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java [hunk 1]
```diff
@@ -545,9 +520,10 @@ public long nextId()
      */
     public void freeId( long id )
     {
-        if ( idGenerator != null )
+        IdGenerator generator = this.idGenerator;
```
Comment (lutovich) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java:523]: Why do we need to store `idGenerator` field to local variable?

Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java:523]: So the value we null-check and the object we call a method on are always the same.

Comment (lutovich) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java:523]: I think `idGenerator` field is only assigned during startup and is never modified concurrently with transaction threads. Isn't it true?

Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java:523]: It's also assigned in `deleteIdGenerator`, which is called during recovery. Plus, it was a memory load. I don't think it's a problem in practice.

**Pull request merged**